inductiva/tasks/run_simulation.py

```diff
@@ -39,4 +45,43 @@ def run_simulation(
         raise RuntimeError(
             f"Expected result to be a string with task_id, got {type(task_id)}")
 
+    if format_utils.getenv_bool("SAVE_SIM_METADATA", True):
+        metadata = {
+            **{
+                "machine_group_id": resource_pool_id,
+                "task_id": task_id,
+                "input_dir": str(input_dir),
+                "storage_dir": str(storage_dir),
+            },
+            **kwargs,
+        }
+        if extra_sim_metadata_to_save is not None:
+            metadata = {**metadata, **extra_sim_metadata_to_save}
+        _save_metadata(save_sim_metadata_dir, metadata)
+
     return task
+
+
+def _save_metadata(save_sim_metadata_dir, metadata):
+    """Save metadata to a .json file.
+
+    Args:
+      save_sim_metadata_dir: Directory where to save the metadata. If
+      `None` it saves on the current working directory.
+      metadata: Dictionary with the metadata to be saved.
+
+    """
+    if save_sim_metadata_dir is None:
+        save_sim_metadata_dir = pathlib.Path().cwd()
+    # Ensure that it is a pthlib object in case the user passed a string.
+    save_sim_metadata_dir = pathlib.Path(save_sim_metadata_dir).resolve()
+
+    task_id = metadata["task_id"]
+    save_sim_metadata_dir /= f"{task_id}"
+    save_sim_metadata_dir.mkdir(parents=True, exist_ok=True)
+    file_path = save_sim_metadata_dir / "sim_metadata.json"
+
+    with open(file_path, "w", encoding="utf-8") as f:
+        json.dump(metadata, f)
+
+    logging.info("Simulation metadata logged to: %s", file_path)
```

Comment:
This can be simplified because we're appending to the same file and the filename is hard coded.

```suggestion
    TASK_METADATA_FILENAME = "task_metadata.json"
    file_path = pathlib.Path().cwd() / TASK_METADATA_FILENAME

    with open(file_path, "a", encoding="utf-8") as f:
        json.dump(metadata, f)
```

this will produce a file like:

```json
# tmp.json
{"a": "1", "b": "2"}
{"a": "3", "b": "4"}
```

that can easily be manipulated with `jq`:

```bash
$ jq '.a' tmp.json
"1"
"3"
```